src/python_testing/test_testing/TestDecorators.py

```diff
@@ -0,0 +1,291 @@
+#
+#    Copyright (c) 2024 Project CHIP Authors
+#    All rights reserved.
+#
+#    Licensed under the Apache License, Version 2.0 (the "License");
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+#
+
+
+# Hooks:
+# If this is a per-endpoint test
+# - If test is run, hook object will get one test_start and one test_stop call per endpoint on which the test is run
+# - If the test is skipped, hook object will get test_start, test_skipped, test_stop
+# If this is a whole-node test:
+# - test will always be run, so hook object will get test_start, test_stop
+#
+# You will get step_* calls as appropriate in between the test_start and test_stop calls if the test is not skipped.
+
+import os
+import sys
+
+import chip.clusters as Clusters
+from chip.clusters import Attribute
+from chip.clusters import ClusterObjects as ClusterObjects
+
+try:
+    from matter_testing_support import (MatterBaseTest, async_test_body, get_accepted_endpoints_for_test, has_attribute,
+                                        has_cluster, per_endpoint_test, per_node_test)
+except ImportError:
+    sys.path.append(os.path.abspath(os.path.join(os.path.dirname(__file__), '..')))
+    from matter_testing_support import MatterBaseTest, async_test_body, has_cluster, has_attribute, get_accepted_endpoints_for_test, per_endpoint_test, per_node_test
```

Comment:
nit, Is there a reason why you import different things when on ImportError?

Reply:
okie dokie - updated.